File: scripts/extract-svg.js
import { readFileSync, writeFileSync } from 'fs';

const html = readFileSync('/vercel/share/v0-project/index.html', 'utf-8');

const svgStart = html.indexOf('<svg xmlns=');
const svgEnd = html.indexOf('</svg>', svgStart) + '</svg>'.length;

if (svgStart === -1 || svgEnd <= svgStart) {
  console.error('Could not find SVG in index.html');
  process.exit(1);
}

const svgContent = html.substring(svgStart, svgEnd);
writeFileSync('/vercel/share/v0-project/public/moscow.svg', svgContent, 'utf-8');
console.log(`Extracted SVG (${svgContent.length} chars) to public/moscow.svg`);

File: scripts/extract_svg.py
import re

with open("/vercel/share/v0-project/index.html", "r", encoding="utf-8") as f:
    html = f.read()

# Find the SVG tag embedded in index.html
match = re.search(r'(<svg xmlns="http://www.w3.org/2000/svg".*?</svg>)', html, re.DOTALL)

if match:
    svg_content = match.group(1)
    with open("/vercel/share/v0-project/public/moscow.svg", "w", encoding="utf-8") as out:
        out.write(svg_content)
    print(f"Extracted SVG: {len(svg_content)} characters")
else:
    print("ERROR: SVG not found in index.html")
